src/WebJobs.Script.Grpc/Channel/GrpcWorkerChannel.cs

```diff
@@ -395,7 +397,7 @@ internal void WorkerInitResponse(GrpcEvent initEvent)
             UpdateCapabilities(_initMessage.Capabilities);
 
             _isSharedMemoryDataTransferEnabled = IsSharedMemoryDataTransferEnabled();
-            _cancelCapabilityEnabled = !string.IsNullOrEmpty(_workerCapabilities.GetCapabilityState(RpcWorkerConstants.HandlesInvocationCancelMessage));
+            _cancelCapabilityEnabled ??= !string.IsNullOrEmpty(_workerCapabilities.GetCapabilityState(RpcWorkerConstants.HandlesInvocationCancelMessage));
```

Comment:
Also, this may not be part of this PR but can we update the name `_cancelCapabilityEnabled` to may be `_invocationCancelCapabilityEnabled` or something on similar lines. Only cancel capability is not communicating the meaning till I saw `RpcWorkerConstants.HandlesInvocationCancelMessage`

Reply:
Currently, host code(in various places) checks the existence of the capability instead of the value being truthy. I agree, Ideally it should be a truthy check. but that might involve coordinating with workers to make sure they send the correct value.

Currently dotnet worker sends capability when the value is truthy. Ex: https://github.com/Azure/azure-functions-dotnet-worker/blob/14cc88859a1b7bbf68ee43409a1ad45a6e8362cb/src/DotNetWorker.Grpc/GrpcWorker.cs#L222-L229